gsoc2018-bharat/src/MakeDictionary.py

```diff
@@ -0,0 +1,54 @@
+import sys
+import re
+import os
+import codecs
+
+def upcase_first_letter(s):
+	return s[0].upper() + s[1:]
+
+# Replaces the Gensim class LineSentence
+class MySentences(object):
+	def __init__(self, dirname):
+		self.dirname = dirname
+
+	def __iter__(self):
+		for root, dirs, files in os.walk(self.dirname):
+			for filename in files:
+				file_path = root + '/' + filename
+				for line in codecs.open(file_path, 'r', encoding='utf-8', errors='ignore'):
+					if line == "\n":
+						continue
+					yield line
+
+def makeDictionary(directory):
+	"""
+	All the links present in the corpus are treated as resources and
+	the text is treated as the surface form for that very entity.
+	"""
+	counter = 0
+	print("Opened ", directory)
+	sentences = MySentences(directory)
+	dictionary = {}
+	for line in sentences:
+		counter += 1
+
+		# Finding all the anchor text, that is the <a> tags
```

Comment:
Isolate the operations you do to a line in its own function.

Reply:
Yes that makes sense. I will clean the code a bit more.